### PR TITLE
Fix downloading cluster specific kubectl

### PR DESCRIPTION
**Pull request by Nokel81**
- The normalization of the architecture was incorrect as the kubectl
  download server uses a different normalization than what OpenLens does
  for its bundled binaries.

- Made that split explicit by introducing some new injectables

- Made the Kubectl class not depend on any global shared state by
  introducing new injectables

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Review by jim-docker — COMMENTED** [review]
The downloading is working but Lens is still using the bundled version in the terminal:

Screen Shot 2022-05-16 at 3 39 55 PM

See the [SHELL-SESSION] error

**Comment by jim-docker** [comment]
> 
> See the [SHELL-SESSION] error

Actually, this error is also in the latest master (it shows every 30s)

**Comment by Nokel81** [comment]
> Actually, this error is also in the latest master (it shows every 30s)

Okay that should be fixed then. But it does look like we are not picking the correct `kubectl` version anyway. I will look into this.

**Comment by Nokel81** [comment]
@jim-docker Fixed.

**Comment by Nokel81** [comment]
Also, how did you encounter that other issue, I have not seen it.

**Comment by jim-docker** [comment]
> Also, how did you encounter that other issue, I have not seen it.

I see it on all 3 platforms, I have a terminal in the dock, which opens when connecting to a cluster. Error appears every ~30s

File: src/main/kubectl/kubectl.ts [hunk 1]
```diff
@@ -299,7 +299,7 @@ export class Kubectl {
       "elif test -f \"$HOME/.profile\"; then",
       "  . \"$HOME/.profile\"",
       "fi",
-      `export PATH="${binariesDir}:${kubectlPath}:$PATH"`,
+      `export PATH="${kubectlPath}:${binariesDir}:$PATH"`,
```
Comment (jim-docker) [inline src/main/kubectl/kubectl.ts:302]: these changes don't help on Windows (it still uses the bundled kubectl)

Comment (Nokel81) [reply src/main/kubectl/kubectl.ts:302]: Fixed

**Comment by Nokel81** [comment]
Closing in favour of https://github.com/lensapp/lens/pull/5413, because master is diverging too far

Will make a PR to merge the release branch back into master once 5.5 is out

**Comment by Nokel81** [comment]
Marking this PR as `skip-changelog` so that it isn't displayed as part of the change-log. We need this PR here because the other PR is targeting the `release/v5.5` branch.

**Review by jim-docker — APPROVED** [review]

**Pull request merged**